File: app/src/main/java/com/morse/ganio/entries/BaseResponse.java
package com.morse.ganio.entries;

import java.io.Serializable;
import java.util.List;

public class BaseResponse<T> implements Serializable {

    private List<T> results;
    private String error;
    private int code;

    public List<T> getResults() {
        return results;
    }

    public void setResults(List<T> results) {
        this.results = results;
    }

    public String getError() {
        return error;
    }

    public void setError(String error) {
        this.error = error;
    }

    public int getCode() {
        return code;
    }

    public void setCode(int code) {
        this.code = code;
    }

    @Override
    public String toString() {
        return "BaseResponse{" +
                "data=" + results +
                ", msg='" + error + '\'' +
                ", code=" + code +
                '}';
    }
}

File: app/src/main/java/com/morse/ganio/entries/BaseResponse.kt
package com.morse.ganio.entries

import java.io.Serializable

data class BaseResponse<T>(var results: List<T>?, var error: Boolean?, var code: Int) : Serializable {

    override fun toString(): String {
        return "BaseResponse{" +
                "data=" + results +
                ", msg='" + error + '\''.toString() +
                ", code=" + code +
                '}'.toString()
    }
}
